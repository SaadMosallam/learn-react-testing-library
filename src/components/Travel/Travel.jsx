import React from "react";
import airPlane from "src/assets/airplane.svg";
const Travel = () => {
  return (
    <div className="container">
      <div className="row">
        <div className="col-12 d-flex justify-content-center mt-5">
          <div
            className="card text-center p-3"
            style={{ width: 400, height: 400 }}
          >
            <img src={airPlane} alt="air plane" />
            <h4>Travel Anywhere</h4>
            <p className="p-1">
              Our premium package allows you to take exotic trips anywhere at
              the cheapest prices!!
            </p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Travel;
