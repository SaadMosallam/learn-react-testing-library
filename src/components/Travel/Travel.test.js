import { render } from '@testing-library/react';
import React from 'react';
import Travel from './Travel';

describe('Travel tests', () => {
    test('Travel component rendered correctly', () => {
        const { container } = render(<Travel />);
        // we can use either toMatchSnapshot or toMatchInlineSnapshot and provide the snapshot as text
        expect(container).toMatchSnapshot();
    });
});