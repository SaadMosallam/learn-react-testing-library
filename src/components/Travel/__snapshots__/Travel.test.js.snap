// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Travel tests Travel component rendered correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-12 d-flex justify-content-center mt-5"
      >
        <div
          class="card text-center p-3"
          style="width: 400px; height: 400px;"
        >
          <img
            alt="air plane"
            src="airplane.svg"
          />
          <h4>
            Travel Anywhere
          </h4>
          <p
            class="p-1"
          >
            Our premium package allows you to take exotic trips anywhere at the cheapest prices!!
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
